spacy/cli/package.py

```diff
@@ -113,8 +114,18 @@ def package(
             msg.fail(
                 f"Model name ('{name}') is not a valid python identifier. "
                 "This is required to correctly package the pipeline so it can be imported as a module.",
-                "Valid characters for identifiers are ASCII A-Z, a-z, _ (underscore) "
-                "and 0-9 (except as the first character)",
+                "We recommend names that use ASCII A-Z, a-z, _ (underscore), "
+                "and 0-9 (except as the first character). "
+                "For specific details see: https://docs.python.org/3/reference/lexical_analysis.html#identifiers",
+                exits=1,
+            )
+        if not _is_permitted_package_name(name):
+            msg.fail(
+                f"Model name ('{name}') is not a permitted package name. "
+                "This is required to correctly package the pipeline.",
+                "We recommend names that use ASCII A-Z, a-z, _ (underscore), "
+                "and 0-9, and start with a character. "
+                "For specific details see: https://www.python.org/dev/peps/pep-0426/#name",
```

Comment:
The restriction about the first character don't apply because this is concatenated with `lg_` for the final package name.

Reply:
Good point. That actually makes me think we should check this _after_ the language is concatenated to the model name, rather than try and incorporate an exception to these rules to account for the fact that the language will eventually be added.

Reply:
I'm going to leave the errors where they are because it'll be much simpler to maintain. Otherwise we'll have to go into too much detail trying to communicate what happens to a package name -- e.g. they could name a model `_leading_underscore` and have it be magically fixed by our naming scheme (it would be come `lang_leading_underscore`), but I think we should encourage the best practice here up front.